typescript/lib/config.ts

```diff
@@ -254,32 +282,20 @@ export class AIConfigRuntime implements AIConfig {
         aiConfigObj.prompts = prompts;
       }
 
-      const defaultFilePath =
-        mode === "yaml" ? "aiconfig.yaml" : "aiconfig.json";
-      if (!filePath) {
-        filePath = this.filePath ?? defaultFilePath;
-      }
-
-      if (mode == null) {
-        if (isYamlExt(filePath)) {
-          mode = "yaml";
-        } else {
-          // Default to JSON
-          mode = "json";
-        }
+      if (aiConfigObj["$schema"] == null) {
+        // Add the $schema property to the JSON object before saving it. In the future this can respect the version specified in the AIConfig.
+        aiConfigObj["$schema"] = "https://json.schemastore.org/aiconfig-1.0";
       }
 
       // TODO: saqadri - make sure that the object satisfies the AIConfig schema
       let aiConfigString;
       if (mode === "yaml") {
         aiConfigString = yaml.dump(aiConfigObj, { indent: 2 });
       } else {
-        // Add the $schema property to the JSON object before saving it. In the future this can respect the version specified in the AIConfig.
```

Comment:
Oh, interesting - this was only done in the JSON format before. Curious if it was intended to be left out in the YAML or just accidentally missed before (and fixed here)?